LIHADOOP-23924 Create dashboard of Dr Elephant using ember

nntnag17 opened the pull request:
This is the pull request for the dashboard of new doctor elephant user interface. The new dashboard has following new features:

1) Cleaner interface
2) Separate tabs for flow level, job level and application level
3) Separate tabs for each user, a user can add or delete new users. 


Dashboard: Application page

![dashboard_application](https://cloud.githubusercontent.com/assets/4449496/19298504/cd7452c8-906b-11e6-9170-3ca58ee70776.png)

Dashboard: Workflow page

![dashboard2](https://cloud.githubusercontent.com/assets/4449496/19298505/cd75fc36-906b-11e6-93a0-7928b3d98591.png)

Dashboard:

![newdashboard](https://cloud.githubusercontent.com/assets/4449496/19267295/9cbdb0fc-8fcb-11e6-8d0f-8dc5be0b8ad7.png)


Tests included: 
Integration tests: 

        web/tests/integration/components/aggregated-metrics-test.js
        web/tests/integration/components/heuristics-summary-test.js
        web/tests/integration/components/jobs-severity-test.js
        web/tests/integration/components/single-application-test.js
        web/tests/integration/components/single-flow-test.js
        web/tests/integration/components/single-job-test.js
        web/tests/integration/components/single-tab-test.js
        web/tests/integration/components/tasks-severity-test.js
        web/tests/integration/components/user-tabs-test.js

Unit tests:

        web/tests/unit/helpers/eq-test.js
        web/tests/unit/helpers/get-bootstrap-severity-code-test.js
        web/tests/unit/helpers/get-color-for-severity-test.js
        web/tests/unit/helpers/get-date-test.js
        web/tests/unit/helpers/get-duration-breakdown-test.js
        web/tests/unit/helpers/get-percentage-test.js
        web/tests/unit/helpers/get-resource-in-gbhours-test.js
        web/tests/unit/helpers/not-empty-test.js
        web/tests/unit/helpers/url-encode-test.js

aalpizar commented on web/app/components/user-tabs.js:
```diff
@@ -0,0 +1,31 @@
+
+    /** brings focus to the input box **/
+    showInput: function() {
+      Ember.$("#user-input").attr("type","text");
```
You don't need to use jQuery to 'focus-in' or 'focus-out' of an input
In the template you can leverage the Ember inputs and assign an action to the 'focus-out' or 'focus-in' event

https://guides.emberjs.com/v2.6.0/templates/input-helpers/

nntnag17 replied:
Done. I'm now using autofocus property of input field.

aalpizar commented on web/app/controllers/dashboard.js:
```diff
@@ -0,0 +1,27 @@
+
+    /** add a new tab and click on it once the rendering is complete **/
+    addTab: function(user) {
+      this.users.addToUsername(user);
```
You can leverage ES6 symantics here by saying
```
addTab(user) {
...
}
```

nntnag17 replied:
Thanks for this info. I've changed the actions to use ES6 semantics.

aalpizar commented on web/app/components/single-tab.js:
```diff
@@ -0,0 +1,10 @@
+
+export default Ember.Component.extend({
+  actions: {
+    /** bubble this action up **/
```
In general, best practices are to document the code. A good starting point would be with JSDocs

http://usejsdoc.org/about-getting-started.html

nntnag17 replied:
Added comments wherever it was missing.

nntnag17 commented on web/app/templates/dashboard/app.hbs:
```diff
@@ -0,0 +1,7 @@
+{{!-- The user tabs. Tabs can be added and deleted --}}
+{{user-tabs deleteTab="deleteTab" setTab="changeTab" addTab="addTab" usernames=model.usernames}}
```
Done.

nntnag17 commented on web/app/templates/dashboard.hbs:
```diff
@@ -0,0 +1,37 @@
+                    <h3>Hello there, I've been busy!</h3>
+
+                    <p>I looked through <b>150</b> jobs today.<br>
+                        About <b>70</b> of them could use some tuning.<br>
```
Done.

nntnag17 commented on web/app/templates/dashboard.hbs:
```diff
@@ -0,0 +1,37 @@
+                <div class="bubble">
+                    <h3>Hello there, I've been busy!</h3>
+
+                    <p>I looked through <b>150</b> jobs today.<br>
```
These values will eventually come from a model. I haven't created the model yet hence have hardcoded it.

nntnag17 commented on web/app/templates/application.hbs:
```diff
@@ -0,0 +1,38 @@
+<style type="text/css">
```
Thanks for this information. I've changed it.

nntnag17 commented on web/app/styles/app.css:
```diff
@@ -0,0 +1,607 @@
+.list-group-item {
```
Done. Thanks for letting me know the best practices.

nntnag17 commented on web/app/components/user-tabs.js:
```diff
@@ -0,0 +1,31 @@
+    /** Adds the tab **/
+    addTab: function() {
+      this.sendAction('addTab', this.newUser);
+      Ember.$("#user-input").attr("type","hidden");
```
Done. I'm now using a boolean variable called showInputBox to hide and show the input box.

nntnag17 commented on web/app/components/user-tabs.js:
```diff
@@ -0,0 +1,31 @@
+
+    /** bubble this action up */
+    setTab: function(tabname) {
+      this.sendAction('setTab', tabname);
```
Done. Good to know about closure actions.

nntnag17 commented:
Hey @aalpizar ,

I've addressed your comments and suggestions. Thanks for taking out the time to review this pull request. I really appreciate it. Please let me know if there are any more comments. 

Thanks,
Anant
